mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -1936,6 +1937,100 @@ def instructions(self) -> str:
+@format_docstring(_MODEL_API_DOC)
+class ConversationalRoleAdherence(BuiltInSessionLevelScorer):
+    """
+    Conversational role adherence evaluates whether an AI assistant maintains its assigned
+    role throughout a conversation.
+
+    This scorer analyzes the assistant's responses for role adherence concerns including:
+
+    - Persona consistency (maintaining personality, tone, and character)
+    - Knowledge boundaries (staying within defined expertise)
+    - Behavioral adherence (following role-implied guidelines)
+    - Role acknowledgment (correctly identifying itself when asked)
+    - Boundary maintenance (refusing or redirecting out-of-scope requests)
+
+    Note: User messages containing out-of-scope requests do not make a conversation
+    non-adherent. Only the assistant's actual responses are evaluated.
+
+    You can invoke the scorer directly with a session for testing, or pass it to
+    `mlflow.genai.evaluate` for running full evaluation on a dataset.
+
+    Args:
+        name: The name of the scorer. Defaults to "conversational_role_adherence".
+        role_description: Description of the role the assistant should maintain.
+        model: {{ model }}
+
+    Example (direct usage):
+
+    .. code-block:: python
+
+        import mlflow
+        from mlflow.genai.scorers import ConversationalRoleAdherence
+
+        # Retrieve a list of traces with the same session ID
+        session = mlflow.search_traces(
+            experiment_ids=[experiment_id],
+            filter_string=f"metadata.`mlflow.trace.session` = '{session_id}'",
+            return_type="list",
+        )
+
+        assessment = ConversationalRoleAdherence(
+            role_description="A helpful cooking assistant that provides recipes."
+        )(session=session)
+        print(assessment)  # Feedback with value "yes" or "no"
+
+    Example (with evaluate):
+
+    .. code-block:: python
+
+        import mlflow
+        from mlflow.genai.scorers import ConversationalRoleAdherence
+
+        session = mlflow.search_traces(
+            experiment_ids=[experiment_id],
+            filter_string=f"metadata.`mlflow.trace.session` = '{session_id}'",
+            return_type="list",
+        )
+        result = mlflow.genai.evaluate(
+            data=session,
+            scorers=[
+                ConversationalRoleAdherence(
+                    role_description="A customer service agent for a tech company."
+                )
+            ]
+        )
+    """
+
+    name: str = "conversational_role_adherence"
+    role_description: str = ""
+    model: str | None = None
+    description: str = (
+        "Evaluate whether an AI assistant maintains its assigned role throughout "
+        "a conversation, checking for persona consistency and boundary violations."
+    )
+
+    def __init__(self, /, role_description: str = "", **kwargs):
+        super().__init__(**kwargs)
+        self.role_description = role_description
```

Comment:
can we use the system prompt from the conversation for this instead? if we hear feedback we need an override we can use this approach as well